source/templates.tex

```diff
@@ -143,12 +143,26 @@
 \begin{note}
 A template cannot have the same name as any other
 name bound in the same scope\iref{basic.scope.scope}, except
-that a function template can share a name with non-template
-functions\iref{dcl.fct} and/or function templates\iref{temp.over}.
+that a function template can share a name with \grammarterm{using-declarator}s,
+a type, non-template functions\iref{dcl.fct} and/or function templates\iref{temp.over}.
 Specializations, including partial specializations\iref{temp.spec.partial},
 do not reintroduce or bind names.
 Their target scope is the target scope of the primary template,
 so all specializations of a template belong to the same scope as it does.
+\begin{example}
+\begin{codeblock}
+void f();
+class f;                                    // OK
+namespace N {
+  void f(int);
+}
+using N::f;                                 // OK
+template<typename> void f(long);            // OK
+template<typename> void f(long);            // error: corresponding overload
```

Comment:
This is a redeclaration and should be fine, right?

Reply:
I updated the example, turning all declarations into definitions. Should be better now.